src/meta-srv/src/handler/failure_handler.rs

```diff
@@ -76,17 +78,27 @@ impl HeartbeatHandler for RegionFailureHandler {
 
         let Some(stat) = acc.stat.as_ref() else { return Ok(()) };
 
+        // TODO(LFC): Filter out the stalled heartbeats:
+        // After the region failover is done, the distribution of region is changed.
+        // We can compare the heartbeat info here with the global region placement metadata,
+        // and remove the incorrect region ident keys in failure detect runner
+        // (by sending a control message).
+
         let heartbeat = DatanodeHeartbeat {
```

Comment:
If a datanode contains over 10,000 regions, i wonder that if the heartbeat request is too large.

Reply:
tracked in https://github.com/GreptimeTeam/greptimedb/issues/1606 , will look into it in the future